planner/core/expression_rewriter.go

```diff
@@ -349,6 +349,9 @@ func (er *expressionRewriter) buildSemiApplyFromEqualSubq(np LogicalPlan, l, r e
 		if lCol, ok := l.(*expression.Column); ok && mysql.HasNotNullFlag(lCol.GetType().Flag) && mysql.HasNotNullFlag(rCol.GetType().Flag) {
 			rCol.InOperand = false
 		}
+		if lcon, ok := l.(*expression.Constant); ok && !lcon.Value.IsNull() && mysql.HasNotNullFlag(rCol.GetType().Flag) {
+			rCol.InOperand = false
+		}
```

Comment:
We should check cases like `CorrelatedColumn in Column` as well?

Reply:
Yes, added